docs/Training-Generalization-Learning.md

```diff
@@ -0,0 +1,123 @@
+# Training Generalized Reinforcement Learning agents
+
+Reinforcement learning has a rather unique setup as opposed to supervised and
+unsupervised learning. Agents here are trained and tested on the same exact 
+environment, which is analogous to a model being trained and tested on an 
+identical dataset in supervised learning! This setting results in overfitting; 
+the inability of the agent to generalize to slight tweaks or variations in the 
+environment. This is problematic in instances when environments are randomly 
+instantiated with varying properties. To make agents more robust, we train an 
+agent over multiple variations of the environment. The agent is trained with 
+the intent that it learns to maintain a minimum performance regardless of the 
+environment variant and that it generalizes to maintain this in unseen future 
```

Comment:
Would rephrase: "... and that it generalizes to be robust to future unseen variants of the environment."